src/device-manager/WeaveDataManagementClient.cpp

```diff
@@ -985,7 +985,7 @@ void WdmClient::ClientEventCallback(void * const aAppState, SubscriptionClient::
         WeaveLogDetail(DataManagement, "Client->kEvent_OnSubscriptionTerminated. Reason: %u, peer = 0x%" PRIX64 "\n",
                        aInParam.mSubscriptionTerminated.mReason, aInParam.mSubscriptionTerminated.mClient->GetPeerNodeId());
         pWdmClient->mpSubscriptionClient->AbortSubscription();
-        err = WEAVE_ERROR_INCORRECT_STATE;
+        err = WEAVE_ERROR_INCORRECT_SUBSCRIPTION;
```

Comment:
This error name doesn't make sense.  How can a subscription be "incorrect"?

What exactly is happening in this case statement?  It looks like the subscription client object is telling its user that the subscription has terminated.  This event includes a reason (error) code, and optionally a status code and profile.  This information has to be passed back to the application.

Reply:
Yep, you are right, I should return reason code. Just update the PR, and return reason code to application.

Thanks